test/test_gru_cell.h

```diff
@@ -0,0 +1,165 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <vector>
+
+#include "test/testhelper.h"
+#include "tiny_dnn/tiny_dnn.h"
+
+using namespace tiny_dnn::activation;
+
+namespace tiny_dnn {
+
+TEST(gru, train) {
+  network<sequential> nn;
+  adagrad optimizer;
+  recurrent_layer_parameters params;
+  params.reset_state = false;
+  nn << recurrent_layer(gru(3, 2), 1, params);
+  nn.weight_init(weight_init::xavier());
+
+  vec_t a(3), t(2), a2(3), t2(2);
```

Comment:
same as @karandesai-96 in https://github.com/tiny-dnn/tiny-dnn/pull/834, could implement a class `rnn_test` ? a lot a of duplicated code in tests